pkg/rotation/reconciler.go

```diff
@@ -389,11 +415,11 @@ func (r *Reconciler) processNextItem() bool {
 	spcps, err := r.store.GetSecretProviderClassPodStatus(key.(string))
 	if err != nil {
 		log.Errorf("failed to get spc pod status, error: %+v", err)
-		r.handleError(err, key)
+		return true
```

Comment:
what's the reason for this change?

Reply:
Did some research and updated to use rate limited queueing to still allow retries when the `SecretProviderClassPodStatus` is not found. Also added a comment to describe why we are making the changes.